### DISCUSSION TITLE
3.1.4版本任务实际已经执行成功，但是Task状态在数据库中没有更新， 还是running状态

### DISCUSSION THREAD

**Discussion by LiuHongze19920301** [apache/dolphinscheduler#14385] [Q&A]
我在使用DolphinScheduler 3.1.x版本的过程中遇到一个问题，当worker执行任务已经成功后，Task在数据库中的状态没有更新，还是running状态。通过分析源码，我梳理了消息状态>    变更的流程。当worker执行完任务后，无论执行成功还是失败，worker都会给master发送任务状态变更的消息，这种消息是可以重试的，只有当master回复ack后worker才会从重试消息>    缓存中移除消息。但是我发现消息的缓存是在内存中的，我还没有发现消息持久化的机制。这样的重试机制是不是会导致问题， 如果worker之前发送给master的消息中间出现了不限于 >    网络相关的问题导致失败了，就需要进行消息重试，但是如果worker进程重启了， 缓存在内存中的需要重试的消息就消失了， 从而导致出现了任务状态丢失更新的问题。寻求社区帮助分析下这个问题。

**Comment by zhaohehuhu** [comment]
I encountered this issue
